### PR TITLE
Add Delta Lake version annotations to `Microsoft.Spark.Extensions.Delta` APIs.

### PR DESCRIPTION
**Pull request by AFFogarty**
This PR adds Delta Lake version annotations to all Delta Lake APIs in `Microsoft.Spark.Extensions.Delta`.  Versions were determined according to equivalent annotations found in the [official Delta Lake API docs](https://docs.delta.io/0.6.1/api/scala/io/delta/tables/index.html).

This PR contributes to #618.

### CODE REVIEW

File: src/csharp/Extensions/Microsoft.Spark.Extensions.Delta/Attributes.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
```
Comment (suhsteve) [inline src/csharp/Extensions/Microsoft.Spark.Extensions.Delta/Attributes.cs:1]: can you put the licensing header?

File: src/csharp/Extensions/Microsoft.Spark.Extensions.Delta/DeltaLakeVersions.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿namespace Microsoft.Spark.Extensions.Delta
```
Comment (suhsteve) [inline src/csharp/Extensions/Microsoft.Spark.Extensions.Delta/DeltaLakeVersions.cs:1]: can you put the licensing header?

File: src/csharp/Extensions/Microsoft.Spark.Extensions.Delta/Attributes.cs (hunk 1 above)
Comment (AFFogarty) [reply src/csharp/Extensions/Microsoft.Spark.Extensions.Delta/Attributes.cs]: Oops, fixed.

File: src/csharp/Extensions/Microsoft.Spark.Extensions.Delta/DeltaLakeVersions.cs (hunk 2 above)
Comment (AFFogarty) [reply src/csharp/Extensions/Microsoft.Spark.Extensions.Delta/DeltaLakeVersions.cs]: Oops, fixed.

**Comment by imback82** [comment]
I triggered the rerun of the failing tests.

**Pull request merged**